src/modules/auth_arnacon/arnacon_core.c

```diff
@@ -0,0 +1,1034 @@
+/*
+ * Arnacon Core Authentication Implementation
+ * 
+ * Copyright (C) 2025 Jonathan Kandel
+ */
+
+#include "arnacon_core.h"
+#include "keccak256.h"
+#include "../../core/dprint.h"
+#include "../../core/mem/mem.h"
+#include <curl/curl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <ctype.h>
+#include <errno.h>
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
+
+/* Ethereum zero address constant */
+#define ETH_ZERO_ADDRESS "0x0000000000000000000000000000000000000000"
+
+/**
+ * Initialize the core authentication system
+ */
+int arnacon_core_init(void) {
+    CURLcode res;
+    
+    /* Initialize curl globally */
+    res = curl_global_init(CURL_GLOBAL_DEFAULT);
+    if (res != CURLE_OK) {
+        LM_ERR("Failed to initialize curl: %s", curl_easy_strerror(res));
+        return -1;
+    }
+    
+    LM_INFO("Core authentication system initialized");
+    return 0;
+}
+
+/**
+ * Cleanup the core authentication system
+ */
+void arnacon_core_cleanup(void) {
+    curl_global_cleanup();
+    LM_INFO("Core authentication system cleaned up");
+}
+
+/**
+ * cURL callback function for collecting response data
+ */
+size_t response_write_callback(void *contents, size_t size, size_t nmemb, struct ResponseData *response) {
+    size_t realsize;
+    char *ptr;
+    
+    realsize = size * nmemb;
+    
+    if (response->memory == NULL) {
```

Comment:
This is using another logic as the auth_web3 module. I think this logic is probably better, just noticed it. The auth_web3 module is also not logging any errors here.